drivers/counter/counter_mspm0gxxxx.c

```diff
@@ -0,0 +1,162 @@
+// Copyright (C) 2024 Bang & Olufsen A/S, Denmark
+//
+// SPDX-License-Identifier: LicenseRef-BnOProprietary
+
+#include <ti/driverlib/dl_timerg.h>
+#include <ti/driverlib/dl_timer.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/counter.h>
+
+LOG_MODULE_REGISTER(counter_mspm0gxxxx, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define DT_DRV_COMPAT ti_mspm0gxxxx_counter
+// NOTE: it seems(?) that the clocks support multiple input channels but
+// they must be configured on capture mode
+#define MAX_CHANNELS  1
+
+struct counter_mspm0gxxxx_channel_data {
+	counter_alarm_callback_t callback;
+	void *user_data;
+};
+
+struct counter_mspm0gxxxx_config {
+	struct counter_config_info info;
+	GPTIMER_Regs *timer;
+	const DL_Timer_ClockConfig clock_cfg;
+	void (*interrupt_init_function)(const struct device *dev);
+};
+
+struct counter_mspm0gxxxx_data {
+	const struct counter_mspm0gxxxx_config *config;
+	struct counter_mspm0gxxxx_channel_data *channel_data;
+	DL_Timer_TimerConfig time_cfg;
+};
+
+static int counter_mspm0gxxxx_start(const struct device *dev)
+{
+	const struct counter_mspm0gxxxx_config *cfg = dev->config;
+	const struct counter_mspm0gxxxx_data *data = dev->data;
+
+	DL_Timer_reset(cfg->timer);
+	DL_Timer_enablePower(cfg->timer);
+
+	DL_Timer_setClockConfig(cfg->timer, (DL_Timer_ClockConfig *)&cfg->clock_cfg);
+	DL_Timer_initTimerMode(cfg->timer, (DL_Timer_TimerConfig *)&data->time_cfg);
```

Comment:
Are these casts necessary?

Reply:
the clockconfig cast silences a `const` warning. the data cast can be removed